atc/db/pipeline_lifecycle.go

```diff
@@ -87,3 +89,41 @@ func archivePipelines(tx Tx, conn Conn, lockFactory lock.LockFactory, rows *sql.
 
 	return nil
 }
+
+func (p *pipelineLifecycle) RemoveBuildEventsForDeletedPipelines() error {
+	rows, err := psql.Select("id").
+		From("deleted_pipelines").
+		RunWith(p.conn).
+		Query()
+	if err != nil {
+		return err
+	}
+
+	var idsToDelete []int
+	for rows.Next() {
+		var id int
+		if err := rows.Scan(&id); err != nil {
+			return err
+		}
+		idsToDelete = append(idsToDelete, id)
+	}
+
+	rows.Close()
+
+	for _, id := range idsToDelete {
+		_, err = p.conn.Exec(fmt.Sprintf("DROP TABLE IF EXISTS pipeline_build_events_%d", id))
```

Comment:
I think it's a great idea to not have all the queries run in a transaction, nice!